esp32-wroom-rp/src/gpio.rs

```diff
@@ -126,3 +126,45 @@ where
         self.wait_for_esp_ack();
     }
 }
+
+#[cfg(test)]
+mod gpio_tests {
+    use super::EspControlPins;
+    use crate::gpio::EspControlInterface;
+    use embedded_hal_mock::pin::{
+        Mock as PinMock, State as PinState, Transaction as PinTransaction,
+    };
+    use embedded_hal_mock::MockError;
+    use std::io::ErrorKind;
+
+    #[test]
+    fn gpio_init_sets_correct_state() {
+        let err = MockError::Io(ErrorKind::NotConnected);
+
+        let cs_expectations = [PinTransaction::set(PinState::High)];
```

Comment:
This is very cool

Reply:
Yeah! The list of expectations is so cool...